packages/zoe/src/contracts/multipoolAutoswap/swap.js

```diff
@@ -22,95 +34,95 @@ export const makeMakeSwapInvitation = (
     });
     const {
       give: { In: amountIn },
-      want: { Out: wantedAmountOut },
+      want: {
+        Out: { brand: brandOut },
+      },
     } = seat.getProposal();
-    const { brand: brandIn, value: inputValue } = amountIn;
-    const brandOut = wantedAmountOut.brand;
+    const { brand: brandIn } = amountIn;
 
     // we could be swapping (1) central to secondary, (2) secondary to
     // central, or (3) secondary to secondary
 
     if (isCentral(brandIn) && isSecondary(brandOut)) {
       const pool = getPool(brandOut);
-      const amountOut = pool.getCentralToSecondaryInputPrice(inputValue);
+      const {
+        amountOut,
+        amountIn: reducedAmountIn,
```

Comment:
I think swap should use the actual `amountIn` that the user chose. Doing `getPriceGivenAvailableInput` and then adjusting based on `reducedAmountIn` seems like the wrong thing to do. Was there a reason to do this?

Reply:
Has anyone confirmed that uniswap has this issue?  Am I correct that since most currencies are in microunits, the overage is in the noise?

Reply:
If it is in the noise, and is otherwise added to the fee, then the extra complexity probably is not worth it.